pkg/bff/users.go

```diff
@@ -78,39 +85,29 @@ func (s *Server) Login(c *gin.Context) {
 		}
 
 		// Set the organization ID in the user app metadata
-		user.AppMetadata[OrgIDKey] = org.Id
-		if err = s.auth0.User.Update(*user.ID, user); err != nil {
-			log.Error().Err(err).Msg("could not update user app_metadata")
-			c.JSON(http.StatusInternalServerError, "could not complete user login")
-			return
-		}
+		appdata.OrgID = org.Id
 	} else {
 		// Get the organization for the specified user
-		org, err := s.db.Organizations().Retrieve(c.Request.Context(), orgID)
+		org, err := s.db.Organizations().Retrieve(c.Request.Context(), appdata.OrgID)
 		if err != nil {
-			log.Error().Err(err).Msg("could not retrieve organization for user VASP verification")
+			log.Error().Err(err).Str("orgid", appdata.OrgID).Msg("could not retrieve organization for user VASP verification")
 			c.JSON(http.StatusInternalServerError, "could not complete user login")
 			return
 		}
 
-		// Create the actual VASP table
-		directory := make(map[string]string)
+		// Ensure the VASP record is correct for the user
 		if org.Testnet != nil && org.Testnet.Id != "" {
-			directory[testnet] = org.Testnet.Id
+			appdata.VASPs.TestNet = org.Testnet.Id
 		}
 		if org.Mainnet != nil && org.Mainnet.Id != "" {
-			directory[mainnet] = org.Mainnet.Id
+			appdata.VASPs.MainNet = org.Mainnet.Id
 		}
+	}
 
-		// Ensure the VASP record is correct for the user
-		if vasps, ok := GetVASPs(user.AppMetadata); !ok || !MapEqual(vasps, directory) {
-			user.AppMetadata[VASPsKey] = directory
-			if err = s.auth0.User.Update(*user.ID, user); err != nil {
-				log.Error().Err(err).Msg("could not update user app_metadata")
-				c.JSON(http.StatusInternalServerError, "could not complete user login")
-				return
-			}
-		}
+	if err = s.SaveAuth0AppMetadata(*user.ID, *appdata); err != nil {
+		log.Error().Err(err).Str("user_id", *user.ID).Msg("could not save user app_metadata")
+		c.JSON(http.StatusInternalServerError, "could not complete user login")
+		return
```

Comment:
Since we are now syncing the user app metadata at the end, this protects us from partial updates?

Reply:
There should only be two possible cases:

1. The user is logging in for the first time and an organization is created for them 
2. The user needs the VASP ids updated from their organization

In case #1 - the following data is saved:

```json
{
  "orgid": "uuid",
  "vasps": {
    "testnet": "",
    "mainnet": "",
  }
}
```

Case #1 should only happen once. 

Because I moved the syncing to the end, case #2 is going to happen on every single login since I removed the `MapEqual` check. I debated whether or not we should do this -- it is safer, and logins are infrequent. However, we will eventually need some change detection to alert the front-end that the user needs to login again; so this is likely temporary.